zend/value.cpp

```diff
@@ -1511,12 +1521,22 @@ bool Value::contains(const char *key, int size) const
         // leap out if no 'has_property' function is not set (which should normally not occur)
         if (!has_property) return false;
 
+#if PHP_VERSION_ID < 80000
         // the property must be a zval, turn the key into a value
         Value property(key, size);
 
         // call the has_property() method (0 means: check whether property exists and is not NULL,
         // this is not really what we want, but the closest to the possible values of that parameter)
         return has_property(_val, property._val, 0, nullptr);
+#else
+        int retval = 0;
+        zend_string *tmp_str;
+        tmp_str = zend_string_init(key, size, 0);
+        retval = has_property(Z_OBJ_P(_val), tmp_str, 0, nullptr);
+        zend_string_release(tmp_str);
+        return retval;
```

Comment:
This can probably happen with some way of getting a pointer to the key or something like that but I couldn't find the appropriate PHP macro for it